crates/oxc_linter/src/config/mod.rs

```diff
@@ -0,0 +1,177 @@
+use std::path::PathBuf;
+
+pub mod errors;
+use oxc_diagnostics::{Error, FailedToOpenFileError};
+use phf::{phf_map, Map};
+use serde_json::Value;
+
+use crate::{
+    rules::{RuleEnum, RULES},
+    AllowWarnDeny,
+};
+
+use self::errors::{FailedToParseConfigError, FailedToParseRuleValueError};
+
+pub fn parse_eslint_config(path: PathBuf) -> Result<Vec<RuleEnum>, Error> {
```

Comment:
For readability and self-documentation, I always start with some kind of struct and a `new` method.

In this case `ESLintConfig` + `fn new(path: PathBuf) {}`